content/engine/install/debian.md

```diff
@@ -98,12 +98,11 @@ Docker from the repository.
    sudo apt-get update
    sudo apt-get install ca-certificates curl gnupg
    sudo install -m 0755 -d /etc/apt/keyrings
-   curl -fsSL {{% param "download-url-base" %}}/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
-   sudo chmod a+r /etc/apt/keyrings/docker.gpg
+   curl -fsSL {{% param "download-url-base" %}}/gpg -O /etc/apt/keyrings/docker.asc
```

Comment:
Oh, erm, silly question; would need `sudo curl` now? (or `| sudo tee`)? Looks like `/etc/apt/keyrings` may not be accessible by everyone;

```bash
ls -l /etc/apt/keyrings
total 4
-rw-r--r-- 1 root root 2760 Jan 25 21:02 docker.gpg
```


The `sudo chmod a+r /etc/apt/keyrings/docker.gpg` was added for some cloud systems where permissions were not allowing traversing the directories, causing things to fail; https://github.com/docker/docs/pull/17070#discussion_r1161130453

Reply:
^ @dvdksn @tianon

Reply:
ah, yes. Would `sudo curl` do or do we need tee?

Reply:
`sudo curl` is probably the easy one, but thinking if cURL depends on user-directories (therefore `sudo` potentially looking for config in `root`'s home-dir and such. running cURL as root may have a slightly bigger attack surface (vs `tee`)  as well, but perhaps that's just looking for issues.

I'm sure @tianon has opinions as well if `curl | sudo tee` is preferred over `sudo curl`

Reply:
Yeah, attack surface is all that came to mind for me (generally, running `curl` as root is probably safe, but not the best idea).  That being said, if we use `tee`, we should probably also send the output to `/dev/null`, so I'd personally think `sudo curl` is probably the simpler answer to keep the docs easier to understand (users that paranoid are hopefully not even using this line and are instead doing things like fetching the key by full fingerprint from elsewhere and/or doing deeper verification after download).

Reply:
Yup, `sudo curl` is definitely easier to grasp, so perhaps it's an ok trade-off (readability over "fully correct").

We should look at the directory permissions though (the sudo chmod a+r /etc/apt/keyrings/docker.gpg)

as I recall there were some real-life scenarios outside of the user's control where things broke without https://github.com/docker/docs/pull/17070